db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/QueryTreeNode.java

```diff
@@ -1408,13 +1424,20 @@ protected final TableDescriptor getTableDescriptor(String tableName,
         if(schema.getUUID()==null)
             return null;
 
+        //If not in session schema, look for local temporary table first.
+        //If tables is not under SESSION schema, don't get it from LCC. Table version is not set there.
```

Comment:
Session schema is for storing global temporary tables.
http://db.apache.org/derby/docs/10.0/manuals/reference/sqlj33.html

However, it seems we never implement global temporary tables correctly. We may want to file a Jira and fix it.